### PR TITLE
fix: support explicit k8s version with unspecified release channel

### LINKED ISSUE #1279: the variable "kubernetes_version" is not usable on module beta-private-cluster and safer-cluster
### TL;DR

Hi, 

When you try to define a value in variable `kubernetes_version` on module safer-cluster. the value isn't used on `plan`.
Case tested with all type of release channel (`UNSPECIFIED` included).

### Expected behavior

I would force to use an available version of GKE and decide when I want to upgrade the master of GKE.

I gave a specific version to do it and set the release channel to `UNSPECIFIED`.

### Observed behavior

When I plan my configuration Terraform take the version latest of the current release channel.

### Terraform Configuration

```hcl
module "gke" {
  source  = "terraform-google-modules/kubernetes-engine/google//modules/safer-cluster"
  version = "21.0.0"

  kubernetes_version     = "1.21.11-gke.1100" #This value is ignored
  release_channel        = "UNSPECIFIED"
```

Configuration:
- Provider: registry.terraform.io/hashicorp/google
  Version: 4.22.0
- Provider: registry.terraform.io/hashicorp/google-beta
  Version: 4.22.0
```


### Terraform Version

```sh
v1.2.1
```

### Additional information

We cannot choose the Kubernetes version we want to deploy or when we want to perform upgrade (or manually on the console but it's not a goal to do this if we use Terraform)

The issue seems similar like this one: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/675

Thanks by advance for your support.

### PR DESCRIPTION
**Pull request by bharathkkb**
fixes #1279

Add support for setting `kubernetes_version` when release_channel is set to `UNSPECIFIED`. Per [API](https://cloud.google.com/kubernetes-engine/docs/reference/rest/Shared.Types/Channel) `UNSPECIFIED` is equal to not setting a channel.  Also adds test to validate a custom specified version is used.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: examples/safer_cluster/main.tf [hunk 1]
```diff
@@ -38,6 +38,18 @@ provider "kubernetes" {
...
 }
 
+// A random valid k8s version is retrived
+// to specify as an explicit version.
+data "google_container_engine_versions" "current" {
+  project  = var.project_id
+  location = var.region
+}
+
+resource "random_shuffle" "version" {
```
Comment (apeabody) [inline examples/safer_cluster/main.tf:48]: nit: You might want to add `version = ">= 2.2.1"` to the Random Provider block for `random_shuffle` https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/745d30bfd823d653d3e4d73607576115ebf098bf/examples/safer_cluster/versions.tf#L31

**Review by apeabody — APPROVED** [review]
LGTM! - Added one suggestion

File: examples/safer_cluster/main.tf (hunk 1 above)
Comment (bharathkkb) [reply examples/safer_cluster/main.tf:48]: Added a constraint for latest major v.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Pull request merged**